Fixes #5187: Fix App crashes when you click on Author Name

joseearias opened the pull request:
**Description (required)**

Fixes #5187

What changes did you make and why?

Changed onAuthorViewClicked method so that it renders the author's profile on the browser. Previously it just crashed the application as showed on the issue this PR fixes.

![Screenshot 2023-04-03 at 11 43 52 AM](https://user-images.githubusercontent.com/37851094/229587392-51438c3f-e4ea-4cbb-bfb7-d6db0e9dea4f.jpeg)

**Tests performed (required)**

Tested ProdDebug on Pixel 5 with API level 31.

**Screenshots (for UI changes only)**

Before: 

![Screenshot 2023-04-03 at 12 03 21 PM](https://user-images.githubusercontent.com/37851094/229590930-8d3eb2e2-db15-4772-b643-b9954d4672e0.jpeg)

After:

![Screenshot 2023-04-03 at 12 03 20 PM](https://user-images.githubusercontent.com/37851094/229590963-7aa41b34-cee5-4d6b-b1e4-69cc380773b2.jpeg)

RitikaPahwa4444 commented on app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:
```diff
@@ -1232,6 +1232,12 @@ public void onAuthorViewClicked() {
             return;
         }
+        if (sessionManager.getUserName() == null) {
+            String userProfileLink = "https://commons.wikimedia.org/wiki/User:" + media.getAuthor();
```
I think this base URL (https://commons.wikimedia.org/wiki/) can be replaced by the constant declared in the app, as used in the other fragments too:
https://github.com/commons-app/apps-android-commons/blob/07616485ae213bff99e41b7a507fc32c2fdbc7b1/app/src/main/java/fr/free/nrw/commons/category/CategoryEditHelper.java#L115

RitikaPahwa4444 commented on app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:
```diff
@@ -1232,6 +1232,12 @@ public void onAuthorViewClicked() {
             return;
         }
+        if (sessionManager.getUserName() == null) {
+            String userProfileLink = "https://commons.wikimedia.org/wiki/User:" + media.getAuthor();
```
Just a note on how the artist names and user names differ: Let's say I create an account with user name `RP1234` and can enter my first and last name. The artist name in this case would be `first name + last name`. So, the user's profile can be located on Commons using `RP1234` but not with `first name + last name`. This is where `getAuthor() ` would fail in the first scenario.

joseearias replied:
Would the solution then would be just changing `media.getAuthor()` for `media.getUser()` if I understood correctly?

nicolas-raoul reviewed (changes requested):
Great points Ritika!

Jose, would you mind applying Ritika's suggestions?

Thanks both :-)

joseearias commented:
Hi! Don't worry. I'll try to make the changes as soon as possible

joseearias commented:
I've changed it so that it's not a hardcoded url but instead the declared constant that was mentioned, and also I changed the method media.getAuthor() for media.getUser(). It's seems to be working fine now.

nicolas-raoul reviewed (commented):
Strangely with this PR, while not logged in, when I tap a user name in Explore the achievements activity opens correctly.

RitikaPahwa4444 commented:
This is indeed very strange! For me, it is opening in a browser:

https://user-images.githubusercontent.com/83745993/230549611-7dc8119a-d9e4-4c83-8e38-78b34637b7e5.mp4

joseearias commented:
For me it is the same, it works as it should be when non-logged-in.

[device-2023-04-07-014235.webm](https://user-images.githubusercontent.com/37851094/230556209-6974b837-3021-4736-99c6-b525315b157f.webm)
